lib/rack/session/abstract/id.rb

```diff
@@ -368,7 +369,9 @@ def force_options?(options)
 
         def security_matches?(request, options)
           return true unless options[:secure]
-          request.ssl?
+          # OK to send a secure token over ssl, or to local host
+          # or if the instance is running behind a proxy that handles ssl
+          request.ssl? || request.host == 'localhost' || @trust_proxy
```

Comment:
I'm a little concerned about `request.hsot == 'localhost'`.

Firstly, there are more than one way to refer to localhost, e.g. `127.0.0.1` or `foo.localhost` (see <https://datatracker.ietf.org/doc/html/rfc2606#section-2>).

Why not use `ssl` for localhost development?

Reply:
In RFC 6265, the spec says that the user agent (browser) should not send secure cookies over an unsecure connection.  The spec then says that the user agent must determine what constitutes a secure connection.  In the MDN documentation [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies), it says that a secure cookie is never sent except over HTTPS or to localhost.  I have verified that if the Chrome browser binds to 127.0.0.1, the secure cookie is not sent.  It must explicitly be "localhost".  I don't know why.  Yes, one could use SSL for development, but in general this is not necessary for development so it adds an unnecessary hassle for the developer to use a secure session cookie.  

The problem applies not only to session cookies but to the other Rails cookies as well.  For those cookies, the current behavior is to send them always for localhost, and perhaps if RAILS_ENV is not production -- I am not exactly sure.  For these cookies, the *undocumented* way to ensure secure cookies are sent is

`ActionDispatch::Cookies::CookieJar.always_write_cookie = true`

And, in production, where the Rails server is typically behind a proxy that does the SSL, this is what must be done if secure cookies are to be used.

But there is no way, other than a monkeypatch, to do the same thing for Rails session cookies.

Conceivably, one could omit the reference to localhost, and just use the trust_proxy flag.